common/src/main/scala/org/apache/celeborn/common/meta/DeviceInfo.scala

```diff
@@ -231,18 +231,22 @@ object DeviceInfo {
       getMountPoint(f._1.getAbsolutePath, mountPointToDeviceInfo.keySet())
     }.foreach {
       case (mountPoint, dirs) =>
-        val deviceInfo = mountPointToDeviceInfo.get(mountPoint)
-        val diskInfo = new DiskInfo(
-          mountPoint,
-          dirs.map(_._1).toList,
-          deviceInfo,
-          conf)
-        val (_, maxUsableSpace, threadCount, storageType) = dirs(0)
-        diskInfo.configuredUsableSpace = maxUsableSpace
-        diskInfo.threadCount = threadCount
-        diskInfo.storageType = storageType
-        deviceInfo.addDiskInfo(diskInfo)
-        retDiskInfos.put(mountPoint, diskInfo)
+        if (mountPoint != "") {
```

Comment:
`mountPoint.nonEmpty?`

Reply:
thx, i will update later

Reply:
BTW, would you mind updating line 230 to

```patch
-    workingDirs.groupBy { f =>
-      getMountPoint(f._1.getAbsolutePath, mountPointToDeviceInfo.keySet())
-    }
+    workingDirs.groupBy { case (dir, _, _, _) =>
+      getMountPoint(dir.getCanonicalPath, mountPointToDeviceInfo.keySet())
+    }
```

we'd better use the canonical path as much as possible, in case the unexpected result
```
scala> new java.io.File("../b").getCanonicalPath
res7: String = /Users/b

scala> new java.io.File("../b").getAbsolutePath
res8: String = /Users/chengpan/../b
```

Reply:
ok